pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -309,12 +309,11 @@ func (p *PodAutoscalerInternal) UpdateFromStatus(status *datadoghqcommon.Datadog
 		if len(status.Horizontal.LastActions) > 0 {
 			p.horizontalLastActions = status.Horizontal.LastActions
 
-			// TODO: Store the recommendations history as well, the last actions miss data
-			for _, recommendation := range status.Horizontal.LastActions {
-				if recommendation.RecommendedReplicas != nil {
-					p.horizontalLastRecommendations = addRecommendationToHistory(recommendation.Time.Time, p.horizontalRecommendationsRetention, p.horizontalLastRecommendations, HorizontalScalingValues{
-						Timestamp: recommendation.Time.Time,
-						Replicas:  *recommendation.RecommendedReplicas,
+			for _, recommendation := range status.Horizontal.LastRecommendations {
```

Comment:
Needs to be out as technically not related to `len(status.Horizontal.LastActions) > 0`